Libraries/LibWeb/HTML/CanvasPattern.h

```diff
@@ -13,33 +14,6 @@
 
 namespace Web::HTML {
 
-class CanvasPatternPaintStyle final : public Gfx::PaintStyle {
-public:
-    enum class Repetition {
-        Repeat,
-        RepeatX,
-        RepeatY,
-        NoRepeat
-    };
-
-    static ErrorOr<NonnullRefPtr<CanvasPatternPaintStyle>> create(CanvasImageSource image, Repetition repetition)
-    {
-        return adopt_nonnull_ref_or_enomem(new (nothrow) CanvasPatternPaintStyle(move(image), repetition));
-    }
-
-    Repetition repetition() const { return m_repetition; }
-
-private:
-    CanvasPatternPaintStyle(CanvasImageSource image, Repetition repetition)
-        : m_image(move(image))
-        , m_repetition(repetition)
-    {
-    }
-
-    CanvasImageSource m_image;
-    Repetition m_repetition { Repetition::Repeat };
-};
-
 class CanvasPattern final : public Bindings::PlatformObject {
```

Comment:
The new tests are failing for me locally in the same way as on CI. I was able to fix the tests by removing the final keyword on this line. Apparently it doesn't play nicely with the `dynamic_cast` used by `as_if()`.

Reply:
That didn't work for me! I'll have to investigate this.

Reply:
Oh you meant `CanvasPatternPaintStyle`, yeah that `final` causes it to not have a vtable which makes `dynamic_cast` fail in `is<T>`. I don't think there's a way this can ever work for those types? Pretty annoying tbh 😅